geomesa-arrow/geomesa-arrow-gt/src/main/scala/org/locationtech/geomesa/arrow/io/DeltaWriter.scala

```diff
@@ -832,14 +974,15 @@ object DeltaWriter extends StrictLogging {
   private class AttributeBatchMerger(
       vector: SimpleFeatureVector,
       transfers: Seq[(Int, Int) => Unit],
-      sort: ArrowAttributeReader
+      sort: ArrowAttributeReader,
+      ordering: Ordering[AnyRef]
     ) extends BatchMerger[AttributeBatchMerger](vector, transfers) {
 
-    private var value: Comparable[Any] = sort.apply(0).asInstanceOf[Comparable[Any]]
+    private var value: AnyRef = sort.apply(0)
```

Comment:
The 'Comparable' drops out 'cause ....  List isn't comparable?

Reply:
yep! that was the error thrown when you tried to dictionary encode a list-type attribute.